engine/api/workflow/process.go

```diff
@@ -497,12 +497,14 @@ func processWorkflowNodeRun(dbCopy *gorp.DbMap, db gorp.SqlExecutor, store cache
 		isRoot = true
 	}
 
-	vcsInfos, errVcs := getVCSInfos(db, store, p, gitValues, n, run, !isRoot)
+	var previousGitRepo string
+	pGitRepo := sdk.ParameterFind(&run.BuildParameters, tagGitRepository)
+	if pGitRepo != nil {
+		previousGitRepo = pGitRepo.Value
+	}
+	vcsInfos, errVcs := getVCSInfos(db, store, p, gitValues, n, run, !isRoot, previousGitRepo)
 	if errVcs != nil {
-		if isRoot {
-			return false, errVcs
-		}
-		log.Error("processWorkflowNodeRun> Cannot get VCSInfos")
+		return false, sdk.WrapError(errVcs, "processWorkflowNodeRun> Cannot get VCSInfos")
```

Comment:
i think return an error will raison an error through the http handler; in this case we should set the workflow in failed status

Reply:
no :)